jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/CoreContextHandler.java

```diff
@@ -106,24 +109,64 @@ public void initializeDefaults(Attributes attributes)
 
                         //noinspection unchecked
                         java.util.Collection<Path> deployablePaths = (java.util.Collection<Path>)value;
-                        Optional<Path> optionalDir = deployablePaths.stream()
-                            .sorted(PathCollators.byName(true))
+                        List<Path> nominatedDirs = deployablePaths.stream()
                             .filter(Files::isDirectory)
                             .filter(p -> FileID.isExtension(p.getFileName().toString(), "d"))
-                            .findFirst();
+                            .sorted(PathCollators.byName(true))
+                            .toList();
 
-                        if (optionalDir.isPresent())
+                        if (nominatedDirs.size() == 1)
                         {
                             ResourceFactory resourceFactory = ResourceFactory.of(this);
-                            Resource resourceDir = resourceFactory.newResource(optionalDir.get());
+                            Resource resourceDir = resourceFactory.newResource(nominatedDirs.get(0));
                             setBaseResource(resourceDir);
+                            nominatedDirDefined = true;
+                        }
+                        else if (nominatedDirs.size() > 1)
+                        {
+                            /* this is possible on filesystems that allow things like:
+                             *
+                             * webapps/foo.d
+                             * webapps/FOO.D
+                             * webapps/FOO.d
+                             *
+                             * We want to only support 1 directory in this case.
+                             */
+                            throw new IllegalArgumentException("More than one nominated directory is not supported: " +
+                                nominatedDirs.stream().map(Path::toString).collect(Collectors.joining(", ", "[", "]")));
+                        }
+
+                        // Normal directories.
+                        List<Path> dirs = deployablePaths.stream()
+                            .filter(Files::isDirectory)
+                            .filter(p -> !FileID.isExtension(p.getFileName().toString(), "d"))
```

Comment:
This looks correct - it is ignoring ".d" files.  Should this also list "D"?

Reply:
There are 3 modes of deployment in this class, 4 total if you include the behaviors from StandardContextHandlerFactory too.

1. If the main-path is an archive file (supporting `*.jar`, `*.war`, or `*.zip`) then that is a complete core webapp, with `lib/`, `classes/`, and `static/` portions.
2. If the main-path is a directory `foo.d` then that is a nominated directory, then that is a complete core webapp, with `lib/`, `classes/`, and `static/` portions.
3. If the main-path is any other kind of directory, then that is a static directory.
4. If the main-path is an XML then that is handled by the StandardContextHandlerFactory, not here.